### PR TITLE
[Intel-MKL] Enable MKL Relu6 op

### PR DESCRIPTION
**Pull request by guizili0**

### CODE REVIEW

**Comment by guizili0** [comment]
@agramesh1

File: tensorflow/core/kernels/mkl_relu_op.cc [hunk 1]
```diff
@@ -1251,6 +1254,76 @@ class MklTanhGradOp : public MklReluGradOpBase<Device, T, eltwise_tanh> {
...
+    GetMklShape(context, src_index, &dnn_shape_src);
+
+    Tensor* dst_tensor = nullptr;
+    void* user_i =
+        static_cast<void*>(const_cast<T*>(src_tensor.flat<T>().data()));
+    MklDnnShape dnn_shape_dst;
+    dnn_shape_dst.SetMklTensor(false);
+    AllocateOutputSetMklShape(context, dst_index, &dst_tensor,
+                              src_tensor.shape(), dnn_shape_dst);
+    void* out_o = static_cast<void*>(dst_tensor->flat<T>().data());
```
Comment (ezhulenev) [inline tensorflow/core/kernels/mkl_relu_op.cc]: What's the point in converting `T*` to `void*` and doing reverse conversion the next line (`user_i` and `out_o`)?

File: tensorflow/core/kernels/mkl_relu_op.cc [hunk 2]
```diff
@@ -1251,6 +1254,76 @@ class MklTanhGradOp : public MklReluGradOpBase<Device, T, eltwise_tanh> {
...
+    Tensor* diff_src_tensor = nullptr;
+
+    MklDnnShape dnn_shape_diff_dst;
+    GetMklShape(context, diff_dst_index, &dnn_shape_diff_dst);
+
+    MklDnnShape dnn_shape_diff_src;
+    dnn_shape_diff_src.SetMklTensor(false);
+    AllocateOutputSetMklShape(context, diff_src_index, &diff_src_tensor,
+                              diff_dst_tensor.shape(), dnn_shape_diff_src);
+    void* out_o = static_cast<void*>(diff_src_tensor->flat<T>().data());
```
Comment (ezhulenev) [inline tensorflow/core/kernels/mkl_relu_op.cc]: Same question about pointers, so much syntactic noise but I don't get why it's needed here,

**Review by ezhulenev — CHANGES_REQUESTED** [review]

File: tensorflow/core/kernels/mkl_relu_op.cc (hunk 1 above)
Comment (guizili0) [reply tensorflow/core/kernels/mkl_relu_op.cc]: Done by code change. thanks.

File: tensorflow/core/kernels/mkl_relu_op.cc (hunk 2 above)
Comment (guizili0) [reply tensorflow/core/kernels/mkl_relu_op.cc]: Done by code change. thanks.

**Review by ezhulenev — DISMISSED** [review]

**Comment by agramesh1** [comment]
@yifeif Can you merge this PR? Thanks.

**Comment by yifeif** [comment]
Sorry this fell off of my radar. Importing now!

**Comment by agramesh1** [comment]
@yifeif thanks.

**Comment by drpngx** [comment]
@yifeif should we pull again?

**Comment by guizili0** [comment]
@ezhulenev I have solve the merge conflict, could you help to review it again? thanks.

**Review by ezhulenev — APPROVED** [review]

**Comment by guizili0** [comment]
@yifeif @drpngx I find feedback/copybara, but I cannot check the detail information, can you help me on this?

**Comment by guizili0** [comment]
@yifeif @drpngx I find feedback/copybara failed, but I cannot check the detail information, can you help me on this?

**Comment by drpngx** [comment]
i think it's an unrelated failure (`control_flow_gpu_test`) fails with some DMA non-variant copy error.

**Pull request merged**